### ISSUE TITLE
Cannot reuse template! twice in one template!

### ISSUE THREAD

**Issue by jquesada2016** [sycamore-rs/sycamore#263]
**Describe the bug**
The following does not work:

```rust
#[component(MyComponent<G>)]
pub fn my_component(condition: bool) -> Template<G> {
    let conditional = if condition { template!("conditional stuff") } else { template!() };

    // cloning conditional doesn't help either

    template! {
        button {
            (conditional)    // <-- this doesn't render
            "More stuff"
            (conditional)    // <-- only this renders
        }
    }
}
```

**Expected behavior**
I expect either referencing a shared template more than once, or cloning to actually duplicate the template, and not just increment the RC. Perhaps adding a `cloned` method that would deeply clone the template?

**Environment**

- Sycamore: master
- Browser: Chromium Edge
- OS: Windows 10

**Comment by jquesada2016** [comment]
@lukechu10, I was checking out the code to implement this, but I noticed a couple things. Using just clone only clones the pointer to the node, we have to use `Node::clone_node_with_deep()` to clone the entire tree, but my question is if we should also clone the event listeners and apply them to the cloned node, or just clone the node with no listeners? This behavior is documented [here](https://developer.mozilla.org/en-US/docs/Web/API/Node/cloneNode)

**Comment by lukechu10** [comment]
I don't think it's possible to clone the node with listeners. Listeners can take ownership or capture mutably and cloning them would break Rust's mutability rules. We could wrap all the listeners in `Rc`s but that would add runtime overhead.

I think adding a `clone_deep` method to `Template` is fine but in most cases, it's better to use components to render the same thing multiple times.

**Comment by jquesada2016** [comment]
In this case, since the `EventHandler` is a type alias for `dyn Fn(Event)` and `event` requires a lifetime of `'static`,  there really wouldn't be an issue cloning them. But I completely agree with not doing so now that you mention, because more than anything else, I think it would not be what the user would expect, since it would cause the handler when run to call a function that was more often than not, to trigger a `Signal`, or other side effect, that belongs somewhere else.

Also, if we did add a `clone_deep` method, this wouldn't be super useful in most cases, since this would just give us the `Template` and no ergonomic way to re-add event listeners. Adding a not on clonability for `Template` would be really useful so users don't face the same unexpected behavior (at least to me) as I did. If I have a chance, I'll write something up later today. This issue should be closed once those docs are added.

Reusing components is definitely the better and less complicated route.
